### PR TITLE
Feature/low hanging fruit

### PR DESCRIPTION
**Pull request by mszenher**
Please review after https://github.com/mdsol/delayed_job_sqs/pull/5 is merged.

All but 17 specs now passing.  Most of those failures are due to the fact that job priority isn't implemented yet.

### CODE REVIEW

**Comment by HonoreDB** [comment]
This all looks sane to me, @yzhangmedidata might want to review after merge conflict is resolved.

**Comment by mszenher** [comment]
@HonoreDB @yzhangmedidata Conflicts resolved.  Ready to merge.

File: lib/delayed/backend/sqs.rb [hunk 1]
```diff
@@ -64,6 +73,10 @@ def payload_object=(object)
             @payload_object = object
             self.handler = object.to_yaml
           end
+        rescue TypeError, LoadError, NameError, ArgumentError => e
+          # If we have trouble serializing the object, simply assume it is already serialized and store it as is
```
Comment (yzhangmedidata) [inline lib/delayed/backend/sqs.rb:78]: Seems the error should be logged

File: lib/delayed/backend/actions.rb [hunk 2]
```diff
@@ -25,14 +25,32 @@ def db_time_now
...
+            end
           end
           []
         end
 
+        # Returns true if the job encoded in a message is suitable to be run by a worker; otherwise, return false.
+        def is_job_suitable?(job)
+          job.failed_at.nil? && (job.run_at.nil? || job.run_at <= db_time_now)
+        rescue TypeError, JSON::ParserError => e
+          puts "Cannot parse job description #{job_message} to determine if it is suitable because #{e.message}."
```
Comment (yzhangmedidata) [inline lib/delayed/backend/actions.rb]: Could also log <code>e.class</code>

**Comment by mszenher** [comment]
@yzhangmedidata Comments addressed.  Ready for final review and merge.

**Comment by yzhangmedidata** [comment]
Looks good

**Pull request merged**